collectors/modality-probe-collector-common/src/lib.rs

```diff
@@ -283,320 +354,414 @@ impl TryFrom<&LogFileRow> for LogEntry {
 }
 
 pub fn add_log_report_to_entries(
-    log_report: &LogReport,
+    log_report: &Report,
     session_id: SessionId,
-    initial_segment_id: SegmentId,
     receive_time: DateTime<Utc>,
-    log_entries_buffer: &mut Vec<LogEntry>,
-) -> u32 {
-    let mut next_segment_id = initial_segment_id.0;
+    log_entries_buffer: &mut Vec<ReportLogEntry>,
+) {
     let probe_id = log_report.probe_id;
-    for segment in &log_report.segments {
-        let mut segment_index = 0;
+    let sequence_number = log_report.seq_num.into();
+    let mut sequence_index = 0;
 
-        for clock_bucket in &segment.clocks {
-            log_entries_buffer.push(LogEntry {
-                session_id,
-                segment_id: next_segment_id.into(),
-                segment_index,
-                probe_id,
-                data: LogEntryData::LogicalClock(
-                    clock_bucket.id,
-                    clock_bucket.epoch,
-                    clock_bucket.ticks,
-                ),
-                receive_time,
-            });
+    for fc in &log_report.frontier_clocks {
+        log_entries_buffer.push(ReportLogEntry {
+            session_id,
+            sequence_number,
+            sequence_index,
+            probe_id,
+            data: LogEntryData::FrontierClock(*fc),
+            receive_time,
+        });
+        sequence_index += 1;
+    }
 
-            segment_index += 1;
+    for event in &log_report.event_log {
+        log_entries_buffer.push(ReportLogEntry {
+            session_id,
+            sequence_number,
+            sequence_index,
+            probe_id,
+            data: LogEntryData::from(*event),
+            receive_time,
+        });
+        sequence_index += 1;
+    }
+}
+
+impl TryFrom<&[u8]> for Report {
+    type Error = SerializationError;
+    fn try_from(buf: &[u8]) -> Result<Self, Self::Error> {
+        let report = WireReport::new(buf)?;
+        let (epoch, ticks) = modality_probe::unpack_clock_word(report.clock());
+        let id = report.probe_id()?;
+        let mut owned_report = Report {
+            probe_id: id,
+            probe_clock: LogicalClock { id, epoch, ticks },
+            seq_num: report.seq_num(),
+            frontier_clocks: vec![],
+            event_log: vec![],
+        };
+
+        let clocks_len = report.n_clocks() as usize * mem::size_of::<LogicalClock>();
+        let mut probe_id = None;
+        let payload = report.payload();
+        for u32_bytes in payload[..clocks_len].chunks_exact(mem::size_of::<LogEntry>()) {
+            let raw = le_bytes::read_u32(u32_bytes);
+            if probe_id.is_none() {
+                probe_id = ProbeId::new(
+                    unsafe { LogEntry::new_unchecked(raw) }.interpret_as_logical_clock_probe_id(),
```

Comment:
Why are we skipping the checks here?

Reply:
Since we're interpreting raw bytes, this seems like exactly the spot to do the range checks.

Reply:
For sure. I think what's missing here is two things:
1. first that the `LogEntry` had the clock bit set and _then_ interpreting it.
2. `ok_or_else`-ing the `ProbeId::new` call.

Reply:
I think I covered this in c0adc77eab6c64507ba5d6b58460c064b1aa671d